samples/project-anfield/import_cluster_config.py

```diff
@@ -559,6 +604,58 @@ def create_sources(source, environment, node):
                 body.password = password
             if env == "view":
                 body.view_type = view_enum.KVIEWBOX
+
+        elif environment == env_enum.K_HYPERV:
+            prot_source = source.protection_source
+            name = prot_source.name
+            register_info = source.registration_info
+            body = RegisterProtectionSourceParameters()
+            body.endpoint = name
+            body.hyperv_type = "kSCVMMServer"
+            body.environment = env_enum.K_HYPERV
+            body.username = register_info.username
+            body.password = "Fr8shst8rt"  # configparser.get(name, "password")
+            if register_info.throttling_policy:
+                body.throttling_policy = register_info.throttling_policy
+            register_sources(body, name, source_id)
+
+            # Get the imported source id.
+            imported_source_id = source_mapping.get(source_id, None)
+            if imported_source_id and register_info.throttling_policy_overrides:
+                # Check for source registration status.
+                status = None
+                count = 0
+                while status != "kFinished" and count < 10:
+                    sources = cohesity_client.protection_sources.list_protection_sources_root_nodes(
+                        id=imported_source_id
+                    )
+                    status = sources[0].registration_info.authentication_status
+                    time.sleep(30)
+
+                # Create the datastore name and id mapping.
+                datastores = fetch_hyperv_datastore_mapping(imported_source_id)
+                body = UpdateProtectionSourceParameters()
+                overrides = []
+                for datastore in register_info.throttling_policy_overrides:
+                    datastore_id = datastores.get(datastore.datastore_name, "")
+                    if not datastore_id:
+                        ERROR_LIST.append(
+                            "Failed to find datastore '%s' in the source '%s'"
+                            % (datastore.datastore_name, name)
+                        )
+                    datastore.datastore_id = datastore_id
+                    overrides.append(datastore)
+                body.throttling_policy_overrides = overrides
+                try:
+                    result = (
+                        cohesity_client.protection_sources.update_protection_source(
+                            source_mapping[source_id], body
+                        )
+                    )
+                # Ignoring the timeout error, since the source updation is
+                except TimeoutError as e:
+                    pass
```

Comment:
Catch the exception correctly ?

Reply:
Updating the hyperv source is completed within 30 secs still API response time is greater than 60 secs, which throws exception stating failed, even the source is updated.